### PR TITLE
Make EventLoopFuture.wait() unavailable from async

### PR DESCRIPTION
**Pull request by Lukasa**
Motivation:

ELF.wait() waits for a condition variable to become true, which can frequently lead to extremely long waits. This is a bad thing to call on a Swift Concurrency thread, especially as we have ELF.get() which is preferable.

Modifications:

Make ELF.wait() unavailable from async.

Result:

Users are encouraged to use the correct primitive.

### CODE REVIEW

File: Sources/NIOCore/EventLoopFuture.swift [hunk 1]
```diff
@@ -1232,10 +1233,37 @@ extension EventLoopFuture {
     /// threads: it is primarily useful for testing, or for building interfaces between blocking
     /// and non-blocking code.
     ///
+    /// This is also forbidden in async contexts: prefer ``EventLoopFuture/get``.
+    ///
     /// - returns: The value of the `EventLoopFuture` when it completes.
     /// - throws: The error value of the `EventLoopFuture` if it errors.
+    @available(*, noasync, message: "threads can change between suspension points and therefore the thread specific value too")
```
Comment (dnadoba) [inline Sources/NIOCore/EventLoopFuture.swift:1240]:
```suggestion
    @available(*, noasync, message: "threads can change between suspension points and therefore the thread specific value too", renamed: "get()")
```
This is will produce a fixit that will allow user to easily replace it with `.get()`
```
warning: instance method 'wait' is unavailable from asynchronous contexts; threads can change between suspension points and therefore the thread specific value too; this is an error in Swift 6
    try future.wait()
               ^~~~
               get
```
Users still need to add `await` infront but the compiler will nicely provide a fixit for that too.
```
error: expression is 'async' but is not marked with 'await'
    try future.get()
    ^~~~~~~~~~~~~~~~
        await 
```

---

 I understand that we want to encourage users to use `get()` but I'm not sure I quite understand the `message`:

> threads can change between suspension points and therefore the thread specific value too
 
`wait()` doesn't use any thread specific value and AFAIK it is "safe" to use in async context (except deadlocks) but is just not performant.

Comment (Lukasa) [reply Sources/NIOCore/EventLoopFuture.swift:1240]: Yeah that is just an overhang from a previous usage I didn't replace.

**Review by dnadoba — APPROVED** [review]
LGTM!

**Comment by ktoso** [comment]
LGTM :)

**Pull request merged**